### PR TITLE
Change input sha for testing version

### PR DESCRIPTION
**Pull request by khanhntd**
**Description:** <Describe what has changed.>

When we open a PR, the similar behavior is found across the PR. Therefore, we don't need to use the RUN_ID to tag the image but instead, using the COMMIT_ID (or COMMIT SHA). If we did it this way, it will give us two advantages:
* Increase more visibility to the customer and also for the dev  since we publish CI image public.
* Avoid having similar layers image and also wasting name. In this way, it would show us only the final success image.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -432,7 +433,7 @@ jobs:
         id: versioning
         run: |
           # build a version with github run id so that we can distingush each build for integ-test
-          Version="`cat VERSION`-$GITHUB_RUN_ID"
+          Version="`cat VERSION`-$GITHUB_SHA"
```
Comment (jefchien) [inline .github/workflows/CI.yml:436]: Would rather not have the whole SHA. They can get rather long.
```suggestion
          Version="$(cat VERSION)-$(git rev-parse --short HEAD)"
```

File: README.md [hunk 2]
```diff
@@ -95,7 +95,8 @@ The performance test can be conducted by following the [instructions](https://gi
 
 ### Support
 
-For each merged pull request, a corresponding image is pushed to [public.ecr.aws/aws-otel-test/adot-collector-integration-test](https://gallery.ecr.aws/aws-otel-test/adot-collector-integration-test). 
+For each merged pull request, a corresponding image in the naming convention: ```ADOT_COLLECTOR_VERSION - GITHUB_SHA```
```
Comment (jefchien) [inline README.md:98]:
```suggestion
For each merged pull request, a corresponding image with the naming convention of `[ADOT_COLLECTOR_VERSION]-[GITHUB_SHA]```
```

**Comment by jefchien** [comment]
Are we going to remove the existing images that don't follow this format?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (khanhntd) [reply .github/workflows/CI.yml:436]: Interesting. Nice catch for getting the short SHA though.

File: README.md (hunk 2 above)
Comment (khanhntd) [reply README.md]: Fixed it with the newest commit.

**Comment by khanhntd** [comment]
> Are we going to remove the existing images that don't follow this format?

IMO, we should since it would avoid confusion for the previous format though and in assumption, no customers have been used the previous image before except the nightly build team.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -432,7 +433,7 @@ jobs:
         id: versioning
         run: |
           # build a version with github run id so that we can distingush each build for integ-test
-          Version="`cat VERSION`-$GITHUB_RUN_ID"
+          Version="$(cat VERSION)-$(git rev-parse --short HEAD)"
```
Comment (jefchien) [inline .github/workflows/CI.yml:442]:
```suggestion
          version="$(cat VERSION)-$(git rev-parse --short HEAD)"
```
nitpick: Local variable names should be lower snake case. https://google.github.io/styleguide/shellguide.html#naming-conventions

File: README.md [hunk 4]
```diff
@@ -95,7 +95,8 @@ The performance test can be conducted by following the [instructions](https://gi
 
 ### Support
 
-For each merged pull request, a corresponding image is pushed to [public.ecr.aws/aws-otel-test/adot-collector-integration-test](https://gallery.ecr.aws/aws-otel-test/adot-collector-integration-test). 
+For each merged pull request, a corresponding image with the naming convention of:
```
Comment (jefchien) [inline README.md:98]: nitpick: Don't need the colon.

Comment (khanhntd) [reply README.md]: Fixed it in the newest commit.

File: .github/workflows/CI.yml (hunk 3 above)
Comment (khanhntd) [reply .github/workflows/CI.yml]: The version output is using version but either way, I'm good with the nitpick lol

**Review by aateeqi — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -375,12 +375,16 @@ jobs:
 
       - run: ls -R
 
+        # Build the ssm package and manifest by using a version with github short sha as same as e2etest-preparationfor the following reasons:
+        # - Distingush each build for Integration test
+        # - Increase more visibility to the customer and also for the dev since we publish the image for integration test.
+        # - Avoid having similar layers image and only have a final result image.
```
Comment (jefchien) [inline .github/workflows/CI.yml:381]: These reasons don't really apply for the SSM package. I don't think we give customers access and it isn't an image.

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -375,12 +375,16 @@ jobs:
 
       - run: ls -R
 
+        # Build the ssm package and manifest by using a version with github short sha as same as e2etest-preparationfor the following reasons:
```
Comment (jefchien) [inline .github/workflows/CI.yml:378]:
```suggestion
        # Build the ssm package and manifest by using a version with github short sha same as e2etest-preparation for the following reasons:
```

File: .github/workflows/CI.yml (hunk 5 above)
Comment (khanhntd) [reply .github/workflows/CI.yml]: True thought. Reverted to previous comment but add some extra things.

File: .github/workflows/CI.yml (hunk 6 above)
Comment (khanhntd) [reply .github/workflows/CI.yml]: Reasonable. Fixed it in the newest commit.

**Review by jefchien — APPROVED** [review]

**Pull request merged**